internal/core/window.rs

```diff
@@ -390,8 +389,10 @@ struct PopupWindow {
     location: PopupWindowLocation,
     /// The component that is responsible for providing the popup content.
     component: ItemTreeRc,
-    //// Defines the close behaviour of the popup.
+    /// Defines the close behaviour of the popup.
     close_policy: PopupClosePolicy,
+    /// the item that had the focus in the parent window when the popup was opened
+    focus_item: ItemWeak,
```

Comment:
This field in PopupWindow looks like it's the focus item inside the PopupWindow, when it isn't. I suggest to rename it to previous_focus_item or focus_item_in_parent or focus_item_before_popup.